### PR TITLE
Adds a way for Skrell to pull non-psionics into the Srom.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
As per title. Requested by the Skrell team.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/shareddream/dream_entry.dm [hunk 1]
```diff
@@ -19,21 +28,49 @@ var/list/dream_entries = list()
...
+			if(srom_pulled_by)
+				if(has_psi_aug())
+					dizziness += 40
+					confused += 40
+					slurring += 40
+					eye_blurry += 40
+					return_text += " You feel dizzy, confused and weird..."
+				else
+					return_text += " Your augment staves off most of the post-Srom pull symptoms, but you still feel like your mind is clouded."
+				adjustBrainLoss(10)
```
Comment (SleepyGemmy) [inline code/modules/shareddream/dream_entry.dm:66]: does this heal naturally?

Comment (NonQueueingMatt) [reply code/modules/shareddream/dream_entry.dm:66]: yes

**Review by RyverStyx — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

File: code/modules/shareddream/dream_entry.dm [hunk 2]
```diff
@@ -29,18 +48,33 @@ var/list/dream_entries = list()
 		if(istype(bg) || force_wakeup)
 			// If we choose to be asleep, keep sleeping.
 			if(willfully_sleeping && sleeping && stat == UNCONSCIOUS)
-				sleeping = 5
-				return
-			for(var/thing in SSpsi.processing)
-				var/datum/psi_complexus/psi = thing
-				to_chat(psi.owner, SPAN_CULT("You feel the activity of Srom decrease."))
```
Comment (Llywelwyn) [inline code/modules/shareddream/dream_entry.dm:36]: I think these lines were deleted by mistake - should probably be moved to 54 where the admin msg is

**Review by Llywelwyn — CHANGES_REQUESTED** [review]
Srom exit announcement was deleted erroneously when fixing a merge conflict, I think.

File: code/modules/shareddream/dream_entry.dm [hunk 3]
```diff
@@ -29,18 +48,36 @@ var/list/dream_entries = list()
 		if(istype(bg) || force_wakeup)
 			// If we choose to be asleep, keep sleeping.
 			if(willfully_sleeping && sleeping && stat == UNCONSCIOUS)
-				sleeping = 5
-				return
+				if(can_commune() || (!can_commune() && srom_pulled_by))
```
Comment (mikomyazaki) [inline code/modules/shareddream/dream_entry.dm:51]:
```suggestion
				if(can_commune() || srom_pulled_by)
```

Logically we only get to the second part if can_commune is false, don't need to check it again.

File: code/modules/shareddream/dream_entry.dm [hunk 4]
```diff
@@ -19,6 +28,16 @@ var/list/dream_entries = list()
 			to_chat(bg, "<span class='warning'>Whilst in shared dreaming, you find it difficult to hide your secrets.</span>")
 			if(willfully_sleeping)
 				to_chat(bg, "To wake up, use the \"Awaken\" verb in the IC tab.")
+			if(!srom_pulling && can_commune())
+				var/obj/item/grab/G = r_hand
+				if(!G)
+					G = l_hand
+				if(G)
+					var/mob/living/carbon/human/victim = G.affecting
+					if(ishuman(victim) && !isSynthetic(victim) && victim.is_psi_pingable())
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/modules/shareddream/dream_entry.dm:37]: srom pull shouldn't work on synths even if they have the augment, yeah

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**